workflows/moveBalanceHandler.go

```diff
@@ -0,0 +1,130 @@
+package workflows
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-sdk-erdgo/data"
+)
+
+// MoveBalanceHandlerArgs is the argument DTO for the NewMoveBalanceHandler constructor function
+type MoveBalanceHandlerArgs struct {
+	Proxy                      ProxyHandler
+	TxInteractor               TransactionInteractor
+	ReceiverAddress            string
+	TrackableAddressesProvider TrackableAddressesProvider
+	MinimumBalance             *big.Int
+}
+
+// moveBalanceHandler is an implementation that can create move balance transactions that will empty the balance
+//of the existing accounts
+type moveBalanceHandler struct {
+	proxy                      ProxyHandler
+	cachedNetConfigs           *data.NetworkConfig
+	txInteractor               TransactionInteractor
+	trackableAddressesProvider TrackableAddressesProvider
+	receiverAddress            string
+	minimumBalance             *big.Int
+}
+
+// NewMoveBalanceHandler creates a new instance of the moveBalanceHandler struct
+func NewMoveBalanceHandler(args MoveBalanceHandlerArgs) (*moveBalanceHandler, error) {
+	if check.IfNil(args.TrackableAddressesProvider) {
+		return nil, ErrNilTrackableAddressesProvider
+	}
+	if check.IfNil(args.Proxy) {
+		return nil, ErrNilProxy
+	}
+	if check.IfNil(args.TxInteractor) {
+		return nil, ErrNilTransactionInteractor
+	}
+	if args.MinimumBalance == nil {
+		return nil, ErrNilMinimumBalance
+	}
+
+	mbh := &moveBalanceHandler{
+		proxy:                      args.Proxy,
+		txInteractor:               args.TxInteractor,
+		trackableAddressesProvider: args.TrackableAddressesProvider,
+		receiverAddress:            args.ReceiverAddress,
+		minimumBalance:             args.MinimumBalance,
+	}
+
+	var err error
+	mbh.cachedNetConfigs, err = args.Proxy.GetNetworkConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	return mbh, nil
+}
+
+// GenerateMoveBalanceTransactions wil generate and add to the transaction interactor the move
+// balance transactions. Will output a log error if a transaction will be failed.
+func (mbh *moveBalanceHandler) GenerateMoveBalanceTransactions(addresses []string) {
+	for _, address := range addresses {
+		mbh.generateTransactionHandled(address)
+	}
+}
+
+func (mbh *moveBalanceHandler) generateTransactionHandled(address string) {
```

Comment:
why this naming? `generateHandledTransaction` maybe?

Reply:
handled because I've handled the error by logging it :D

Reply:
renamed to `generateTransactionAndHandleErrors`